anvio/taxoestimation.py

```diff
@@ -0,0 +1,1201 @@
+              end_make_list_taxonomy - start_make_list_taxonomy)
+
+        """posthoc_conover=sp.posthoc_conover_friedman(matrix)
+        print("posthoc_conover")
+        print(posthoc_conover)
+        posthoc_miller_friedman=sp.posthoc_miller_friedman(matrix)
+        print("posthoc_miller_friedman")
+        print(posthoc_miller_friedman)
+        posthoc_siegel_friedman=sp.posthoc_siegel_friedman(matrix)
+        print("posthoc_siegel_friedman")
+        print(posthoc_siegel_friedman)
+        posthoc_durbin=sp.posthoc_durbin(matrix)
+        print("posthoc_durbin")
+        print(posthoc_durbin)
+
+        posthoc_nemenyi_friedman=sp.posthoc_nemenyi_friedman(matrix)
+        print("posthoc_nemenyi_friedman")
+        print(posthoc_nemenyi_friedman)"""
+
+    def del_higher_ranked_individu_matrix(self, matrix, matrixlist_position_entry):
+        maxrank = 0
+        i = 0
+        for liste in matrix:
+            summist = sum(list(liste))
+            if summist > maxrank:
+                maxrank = summist
+                dellist = i
+            i += 1
+        del matrix[dellist]
+        del matrixlist_position_entry[dellist]
+        return matrixlist_position_entry, matrix
+
+    def assign_taxonomie_solo_hit(self, taxonomy):
+        if not taxonomy or not taxonomy[0]:
+            return 0
+        assignation = taxonomy[0]
+
+        for s in taxonomy[1:]:
+            for key in s:
+                if s[key] not in assignation.values():
+                    if anvio.DEBUG:
+                        self.run.info('CONSIDER taxonomy','/'.join(assignation.values()))
+                    assignation.pop(key, None)
+        return(assignation)
+
+    def make_dicoidentitielevel(self):
+        dicoidentitielevel = {}
+        dicoidentitielevel[95] = 't_species'
+        dicoidentitielevel[90] = 't_genus'
+        dicoidentitielevel[85] = 't_family'
+        dicoidentitielevel[80] = 't_order'
+        return(dicoidentitielevel)
+
+    def make_list_taxonomy(self, matrix_pident, matrix, list_position_entry,list_position_ribosomal):
+        taxonomy = []
+        bestlist=[]
+        miniscore = 10000
+        i = 0
+        while i < len(matrix):
+            summist = sum(matrix[i])
+            bestident = max(matrix_pident[i])
+            bestSCG=list_position_ribosomal[matrix_pident[i].index(bestident)]
+            if summist == miniscore:
+                taxonomy.append(self.taxonomy_dict[list_position_entry[i]])
+                bestlist.append([bestSCG,bestident])
+            if summist < miniscore:
+                taxonomy = []
+                taxonomy.append(self.taxonomy_dict[list_position_entry[i]])
+                miniscore = summist
+                bestlist=[[bestSCG,bestident]]
+            i += 1
+        return(taxonomy,bestlist)
+
+
+
+    def reduce_assignation_level(self,taxonomy,bestlist):
+        i=0
+
+
+        for possibilitie in taxonomy:
+            j=-1
+
+            cutoff=50
```

Comment:
this line shouldn't be here but in the function header which should look something like this:

```
def reduce_taxonomic_assignment_level(self, current_taxonomy, list_of_best_hits, cutoff=50):
    (...)
```